src/NotificationTarget.php

```diff
@@ -1130,87 +1130,49 @@ final public function addForProfile($profiles_id)
-            $sender['name']  = $CFG_GLPI['from_email_name'];
-        } else {
-            $admin_email      = trim(Entity::getUsedConfig('admin_email', $this->getEntity(), '', ''));
-            $admin_email_name = trim(Entity::getUsedConfig('admin_email_name', $this->getEntity(), '', ''));
-
-            if (NotificationMailing::isUserAddressValid($admin_email)) {
-               //If the entity administrator's address is defined, return it
-                $sender['email'] = $admin_email;
-                $sender['name']  = $admin_email_name;
+        if (!$this->allowResponse()) {
+            $admin = Config::getNoReplyEmail();
+            if ($admin['email'] !== null) {
+                return $admin;
             } else {
-               //Entity admin is not defined, return the global admin's address
-                $sender['email'] = $CFG_GLPI['admin_email'];
-                $sender['name']  = $CFG_GLPI['admin_email_name'];
+                trigger_error('No-Reply address is not defined in notifications configuration.', E_USER_WARNING);
             }
         }
 
-        if (
-            !$this->allowResponse()
-            && isset($CFG_GLPI['admin_email_noreply'])
-            && !empty($CFG_GLPI['admin_email_noreply'])
-        ) {
-           // Override with no reply email if defined
-            $sender['email'] = $CFG_GLPI['admin_email_noreply'];
-
-            if (
-                isset($CFG_GLPI['admin_email_noreply_name'])
-                && !empty($CFG_GLPI['admin_email_noreply_name'])
-            ) {
-               // Override name with no replay name if defined
-                $sender['name']  = $CFG_GLPI['admin_email_noreply_name'];
-            }
+        // No noreply defined, try to use admin email as a replacement
+        $admin = Config::getAdminEmail($this->getEntity());
+        if ($admin['email'] === null) {
+            trigger_error('Admin email address is not defined in notifications configuration.', E_USER_WARNING);
         }
 
-        return $sender;
+        return $admin;
     }
 
 
     /**
      * Get the reply to address
      *
-     * @param $options   array
-     *
-     * @return the reply to address
+     * @return array [email => replyto address, name => replyto name]
      **/
-    public function getReplyTo($options = [])
+    public function getReplyTo(): array
     {
-        global $CFG_GLPI;
-
-       //If the entity administrator's address is defined, return it
-        $admin_reply      = trim(Entity::getUsedConfig('admin_reply', $this->getEntity(), '', ''));
-        $admin_reply_name = trim(Entity::getUsedConfig('admin_reply_name', $this->getEntity(), '', ''));
+        $replyto = Config::getReplyToEmail($this->getEntity());
+        if ($replyto['email'] !== null) {
+            return $replyto;
+        } else {
+            trigger_error('Reply-To address is not defined in notifications configuration.', E_USER_WARNING);
```

Comment:
Not sure this warning is usefull here, and not sure the fallback is usefull here.

The `no-reply` warning seems important to me, as lack of configuration will fallback to the admin email, and we do not want this email to receive response.

Lack of `admin_reply` may be normal, and empty values are already handled correctly in other part of code.

Reply:
I came to the same conclusion, you posted this at the same time as I fixed it 👍 
Waiting for tests run but all should be OK this time